errors/common.go

```diff
@@ -44,6 +45,14 @@ func IsInvalidFormatErr(err error) bool {
 	return IsSameError(errInvalidFormat, err)
 }
 
+func ErrUnknownModule(mod string) TMError {
```

Comment:
TMErrors is like `pkg/errors` that allow us to add ABCI error codes. Those error codes are super important.

It is just an interface and we can pass them around as error everywhere actually.  Just know that they have a hidden error code.

But yes, I think everything in basecoin should only use `basecoin/errors`, not `pkg/errors` or `errors` to keep it clearer.

I think the (Result, error) and `err != nil` is much more idiomatic go than (res Result) and `res.IsErr()`.  For protobuf, we need one object, so they get combined when we hit the abci interface... fair enough.  Internally, we don't have to worry about protobuf, so I build the structs that make more sense for our framework rather than wire everything to the abci protocol.